lib/platform/github/index.ts

```diff
@@ -146,12 +146,27 @@ export async function initPlatform({
 // Get all repositories that the user has access to
 export async function getRepos(): Promise<string[]> {
   logger.debug('Autodiscovering GitHub repositories');
+  const opts = hostRules.find({
+    hostType: PlatformId.Github,
+    url: platformConfig.endpoint,
+  });
   try {
-    const res = await githubApi.getJson<{ full_name: string }[]>(
-      'user/repos?per_page=100',
-      { paginate: 'all' }
-    );
-    return res.body.map((repo) => repo.full_name);
+    const isGHApp = opts.token?.startsWith('x-access-token:');
```

Comment:
I think this detection should happen in `initPlatform` and stored to `platformConfig`. So we can reuse later where required, like `gheVersion`.

https://github.com/renovatebot/renovate/blob/a3635b7d9ebdcfb7c122cc6b8738cd586250b0ea/lib/platform/github/index.ts#L452-L455

@rarkins WDYT?

Reply:
I agree that's a better place to do the detection. I moved it to initPlatform in the latest commits, let's see what @rarkins thinks